java/src/main/native/src/maps_column_view.cu

```diff
@@ -79,16 +79,32 @@ maps_column_view::get_values_for(scalar const &lookup_key, rmm::cuda_stream_view
   return get_values_for_impl(*this, lookup_key, stream, mr);
 }
 
-std::unique_ptr<column> maps_column_view::contains(scalar const &lookup_key,
-                                                   rmm::cuda_stream_view stream,
-                                                   rmm::mr::device_memory_resource *mr) const {
-  CUDF_EXPECTS(lookup_key.type().id() == keys_.child().type().id(),
+template <typename KeyT>
+std::unique_ptr<column> contains_impl(maps_column_view const &maps_view, KeyT const &lookup_keys,
+                                      rmm::cuda_stream_view stream,
+                                      rmm::mr::device_memory_resource *mr) {
+  auto const keys_ = maps_view.keys();
+  CUDF_EXPECTS(lookup_keys.type().id() == keys_.child().type().id(),
                "Lookup keys must have the same type as the keys of the map column.");
-  auto const contains = lists::detail::contains(keys_, lookup_key, stream);
-
+  auto const contains = lists::detail::contains(keys_, lookup_keys, stream, mr);
```

Comment:
This variable is not returned so don't use `mr`.
```suggestion
  auto const contains = lists::detail::contains(keys_, lookup_keys, stream);
```

Reply:
Can you explain a bit more? What is the relation between a variable being returned and using the device_memory_resource passed in?

Reply:
The returned variable needs to be constructed from the current `mr` but the temporary variable shouldn't (instead, it will be constructed from the default memory resource).

Reply:
Since `contains` is not returned, we don't pass in `mr` thus it will be constructed from the default memory resource.